backend/src/model/entity/plants_impl.rs

```diff
@@ -3,65 +3,45 @@
     /// * Unknown, diesel doesn't say why it might error.
-    pub async fn find_all(conn: &mut AsyncPgConnection) -> QueryResult<Vec<PlantsSummaryDto>> {
-        let query_result = plants::table.select(all_columns).load::<Self>(conn).await;
-        query_result.map(|v| v.into_iter().map(Into::into).collect())
-    }
-
-    /// Search all plants whose name or species name contains the user provided query string.
-    /// To save traffic, the maximum number of results is limited.
-    ///
-    /// # Errors
-    /// * Unknown, diesel doesn't say why it might error.
-    pub async fn search(
-        search_term: &String,
-        limit: i32,
-        offset: i32,
+    pub async fn find(
+        search_parameters: PlantsSearchParameters,
+        page_parameters: PageParameters,
         conn: &mut AsyncPgConnection,
-    ) -> QueryResult<PlantsSearchDto> {
-        let query_with_placeholders = format!("%{search_term}%");
-        // Load one additional row to check whether there are more pages available.
-        let limit_plus_one = limit + 1;
-
-        let query = plants::table
-            .filter(
-                binomial_name
-                    .ilike(&query_with_placeholders)
-                    .or(array_to_string(common_name, " ").ilike(&query_with_placeholders)),
-            )
-            .select(all_columns)
-            .order((binomial_name, common_name))
-            .limit(limit_plus_one.into())
-            .offset(offset.into());
-
-        let query_result = query.load::<Self>(conn).await;
-        query_result.map(|v| {
-            let results: Vec<PlantsSummaryDto> = v.into_iter().map(Into::into).collect();
-            let results_len = results.len();
-            // Perform an explicit conversion to make clippy happy.
-            let unsigned_limit = limit.unsigned_abs() as usize;
-            let unsigned_limit_plus_one = limit_plus_one.unsigned_abs() as usize;
-
-            PlantsSearchDto {
-                plants: results.into_iter().take(unsigned_limit).collect(),
-                // If there is at least one more element than the defined limit,
-                // more pages may still be loaded.
-                has_more: results_len == unsigned_limit_plus_one,
-            }
-        })
+    ) -> QueryResult<Page<PlantsSummaryDto>> {
+        let mut query = plants::table.select(all_columns).into_boxed();
+
+        if let Some(term) = search_parameters.name {
+            query = query
+                .filter(
+                    binomial_name
+                        .ilike(format!("%{term}%"))
+                        .or(array_to_string(common_name, " ").ilike(format!("%{term}%"))),
+                )
+                .order((binomial_name, common_name));
+        }
+
+        let query_page = query
+            .paginate(page_parameters.page)
+            .per_page(page_parameters.per_page)
+            .load_page::<Self>(conn)
+            .await;
+        query_page.map(Page::from)
```

Comment:
Maybe write this without using a variable? Both are fine so you can chose either, but it might be easier to just `.await.map(Page::from)` and immediately implicitly return. (Same for other occurences)

Reply:
I'd rather leave it like this as it made debugging the mapping from dto to entity easier when implementing it.
It might help in the future when hunting bugs.